shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/rulealtered/RuleAlteredJobPersistService.java

```diff
@@ -0,0 +1,175 @@
+        Map<Integer, JobPersistIntervalParameter> intervalParamMap = JOB_PERSIST_MAP.getOrDefault(jobId, Collections.emptyMap());
+        JobPersistIntervalParameter parameter = intervalParamMap.get(shardingItem);
+        if (null == parameter) {
+            log.debug("Persist interval parameter is null, jobId={}, shardingItem={}", jobId, shardingItem);
+            return;
+        }
+        parameter.getAlreadyPersisted().compareAndSet(true, false);
+        if ((currentTimeMills - parameter.getPersistTime().get()) < MIN_PERSIST_INTERVAL_MILLIS) {
+            return;
+        }
+        if (parameter.getLock().tryLock()) {
+            try {
+                SINGLE_EXECUTOR.execute(() -> persist(jobId, shardingItem, currentTimeMills, parameter));
+                // CHECKSTYLE:OFF
+            } catch (final RuntimeException ex) {
+                // CHECKSTYLE:ON
+                log.error("Persist job process failed, job id: {}, sharding item: {}", jobId, shardingItem, ex);
+            } finally {
+                parameter.getLock().unlock();
+            }
+        }
+    }
+    
+    private static void persist(final String jobId, final int shardingItem, final long persistTimeMillis, final JobPersistIntervalParameter param) {
+        Optional<RuleAlteredJobScheduler> jobSchedule = getJobSchedule(jobId, shardingItem);
+        if (!jobSchedule.isPresent()) {
+            log.warn("job schedule not exists, job id: {}, sharding item: {}", jobId, shardingItem);
+            return;
+        }
+        log.info("execute persist, job id={}, sharding item={}, persistTimeMillis={}", jobId, shardingItem, persistTimeMillis);
+        REPOSITORY_API.persistJobProgress(jobSchedule.get().getJobContext());
+        param.getPersistTime().set(persistTimeMillis);
+        param.getAlreadyPersisted().set(true);
+    }
+    
+    private static final class PersistJobContextRunnable implements Runnable {
+        
+        @Override
+        public void run() {
+            long currentTimeMillis = System.currentTimeMillis();
+            for (Entry<String, Map<Integer, JobPersistIntervalParameter>> entry : JOB_PERSIST_MAP.entrySet()) {
+                entry.getValue().forEach((shardingItem, param) -> {
+                    if (param.getAlreadyPersisted().get() || currentTimeMillis - param.getPersistTime().get() < MIN_PERSIST_INTERVAL_MILLIS) {
+                        return;
+                    }
+                    boolean tryLock = false;
+                    try {
+                        tryLock = param.getLock().tryLock();
+                        if (tryLock) {
+                            persist(entry.getKey(), shardingItem, currentTimeMillis, param);
+                        }
+                    } finally {
+                        if (tryLock) {
+                            param.getLock().unlock();
+                        }
+                    }
```

Comment:
`tryLock` might be not necessary anymore if `triggerPersist` remove persistance.